### PR TITLE
Refactor Navigation

### PR DESCRIPTION
**Pull request by Tishasoumya-02**
#4460

### CODE REVIEW

File: src/hooks/intl/useIntls.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { useSelector } from 'react-redux';
+
+export function useIntls() {
```
Comment (nileshgulia1) [inline src/hooks/intl/useIntls.js:3]: maybe rename it to useLang?

File: src/components/theme/Navigation/Navigation.jsx [hunk 2]
```diff
@@ -27,176 +23,108 @@ const messages = defineMessages({
...
 
-export default compose(
-  injectIntl,
-  connect(
-    (state) => ({
-      token: state.userSession.token,
-      items: state.navigation.items,
-      lang: state.intl.locale,
+Navigation.propTypes = {
+  getNavigation: PropTypes.func.isRequired,
```
Comment (nileshgulia1) [inline src/components/theme/Navigation/Navigation.jsx:115]: Unnecessary proptype.

File: src/components/theme/Navigation/Navigation.jsx [hunk 3]
```diff
@@ -27,176 +23,108 @@ const messages = defineMessages({
...
     }),
-    { getNavigation },
-  ),
-)(Navigation);
+  ).isRequired,
+  lang: PropTypes.string.isRequired,
+};
+
+Navigation.defaultProps = {
+  token: null,
```
Comment (nileshgulia1) [inline src/components/theme/Navigation/Navigation.jsx:127]: This needs to be removed.

**Review by sneridagh — CHANGES_REQUESTED** [review]
Some small changes. Let me thing about the dispatches.

**Review by nileshgulia1 — APPROVED** [review]

**Review by nileshgulia1 — APPROVED** [review]

**Review by sneridagh — APPROVED** [review]

**Pull request merged**